stl/inc/ranges

```diff
@@ -2490,6 +2490,428 @@ namespace ranges {
+    template <input_range _Vw, forward_range _Pat>
+        requires view<_Vw> && view<_Pat>
+            && indirectly_comparable<iterator_t<_Vw>, iterator_t<_Pat>, _RANGES equal_to>
+            && (forward_range<_Vw> || _Tiny_range<_Pat>)
+    class split_view;
+    // clang-format on
+
+    template <class _Vw, class _Pat>
+    class _Split_view_base : public view_interface<split_view<_Vw, _Pat>> {
+    protected:
+        /* [[no_unique_address]] */ iterator_t<_Vw> _Current{};
+    };
+    template <forward_range _Vw, class _Pat>
+    class _Split_view_base<_Vw, _Pat> : public view_interface<split_view<_Vw, _Pat>> {};
+
+    // clang-format off
+    template <input_range _Vw, forward_range _Pat>
+        requires view<_Vw> && view<_Pat>
+            && indirectly_comparable<iterator_t<_Vw>, iterator_t<_Pat>, _RANGES equal_to>
+            && (forward_range<_Vw> || _Tiny_range<_Pat>)
+    class split_view : public _Split_view_base<_Vw, _Pat> {
+        // clang-format on
+    private:
+        /* [[no_unique_address]] */ _Vw _Range{};
+        /* [[no_unique_address]] */ _Pat _Pattern{};
+
+        template <bool>
+        class _Inner_iter;
+
+        template <class>
+        class _Outer_iter_base {};
+
+        // clang-format off
+        template <forward_iterator _Iter>
+        class _Outer_iter_base<_Iter> {
+            // clang-format on
+        protected:
+            _Iter _Current{};
+
+        public:
+            using iterator_category = input_iterator_tag;
+
+            _Outer_iter_base() = default;
+            constexpr _Outer_iter_base(_Iter _Current_) noexcept(is_nothrow_move_constructible_v<_Iter>)
+                : _Current{_STD move(_Current_)} {}
+        };
+
+        template <bool _Const>
+        class _Outer_iter : public _Outer_iter_base<iterator_t<_Maybe_const<_Const, _Vw>>> {
+        private:
+            template <bool>
+            friend class _Inner_iter;
+            friend _Outer_iter<true>;
+
+            using _Mybase   = _Outer_iter_base<iterator_t<_Maybe_const<_Const, _Vw>>>;
+            using _ParentTy = _Maybe_const<_Const, split_view>;
+            using _BaseTy   = _Maybe_const<_Const, _Vw>;
+
+            _ParentTy* _Parent = nullptr;
+
+            _NODISCARD constexpr iterator_t<_BaseTy>& _Get_current() noexcept {
+                if constexpr (forward_range<_BaseTy>) {
+                    return this->_Current;
+                } else {
+                    return _Parent->_Current;
+                }
+            }
+
+            _NODISCARD constexpr const iterator_t<_BaseTy>& _Get_current() const noexcept {
+                if constexpr (forward_range<_BaseTy>) {
+                    return this->_Current;
+                } else {
+                    return _Parent->_Current;
+                }
+            }
+
+            _NODISCARD constexpr bool _At_end() const
+                noexcept(noexcept(_Implicitly_convert_to<bool>(_Get_current() == _RANGES end(_Parent->_Range)))) {

```

Comment:
Minor nitpick: Can you write a paper that enforces `operator==` to return `bool`?

Reply:
Done: [dxxxx.pdf](https://github.com/microsoft/STL/files/6131623/dxxxx.pdf)

Reply:
Ship it!